src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogAgentClient.java

```diff
@@ -369,25 +368,24 @@ private void flushSafely() throws IOException {
 
     @Override
     public TraceWriteStrategy createTraceWriteStrategy() {
-        DatadogGlobalConfiguration datadogGlobalDescriptor = DatadogUtilities.getDatadogGlobalDescriptor();
-        String urlParameters = datadogGlobalDescriptor != null ? "?service=" + datadogGlobalDescriptor.getCiInstanceName() : "";
-        String url = String.format("http://%s:%d/evp_proxy/v1/api/v2/webhook/%s", hostname, traceCollectionPort, urlParameters);
-
-        Map<String, String> headers = Map.of(
-              "X-Datadog-EVP-Subdomain", "webhook-intake",
-              "DD-CI-PROVIDER-NAME", "jenkins",
-              "Content-Encoding", "gzip");
-        JsonPayloadSender<Payload> payloadSender = new CompressedBatchSender<>(client, url, headers, PAYLOAD_SIZE_LIMIT, p -> p.getJson());
-
-        TraceWriteStrategyImpl evpStrategy = new TraceWriteStrategyImpl(Track.WEBHOOK, payloadSender::send);
-        TraceWriteStrategyImpl apmStrategy = new TraceWriteStrategyImpl(Track.APM, this::sendSpansToApm);
-        return new AgentTraceWriteStrategy(evpStrategy, apmStrategy, this::isEvpProxySupported);
-    }
+        Set<String> agentEndpoints = fetchAgentEndpoints();
+        boolean evpProxyExists = agentEndpoints.contains("/evp_proxy/v3/");
+        if (evpProxyExists) {
+            DatadogGlobalConfiguration datadogGlobalDescriptor = DatadogUtilities.getDatadogGlobalDescriptor();
+            String urlParameters = datadogGlobalDescriptor != null ? "?service=" + datadogGlobalDescriptor.getCiInstanceName() : "";
+            String url = String.format("http://%s:%d/evp_proxy/v1/api/v2/webhook/%s", hostname, traceCollectionPort, urlParameters);
```

Comment:
We're checking `v3` and `v4`, but then we're always sending to `v1`. Could we just drop a comment saying that this is intended, as we discussed offline.

Reply:
Added a comment